### PR TITLE
feat: shared model manager

### PR DESCRIPTION
**Pull request by hrnn**
127 0 0 1_8000_shared_model_manager_

### CODE REVIEW

File: docs/shared_model_manager.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+- `add()`: Registers a machine learning model class with the manager. The actual model instance is not loaded at this point.
+- `fetch_model()`: Retrieves the previously added model class and creates (loads) the actual model instance. This function utilizes PyTorch interface `to`, to handle device (CPU/GPU) allocation based on availability.
+
+
+The usage example demonstrates adding models and then using them with their respective functionalities.
+
+
+```python
+# Add models to the pool
```
Comment (CamiloInx) [inline docs/shared_model_manager.md:16]: @hrnn should we add how to initialize the SharedModelManager class?

File: vision_agent_tools/tools/shared_types.py [hunk 2]
```diff
@@ -1,8 +1,15 @@
+from enum import Enum
 from pydantic import BaseModel
 
 
+class Device(str, Enum):
+    GPU = "cuda"
+    CPU = "cpu"
```
Comment (CamiloInx) [inline vision_agent_tools/tools/shared_types.py:7]: @hrnn @dillonalaird @camiloaz Should we add `mps` for mac users?

File: vision_agent_tools/tools/owlv2.py [hunk 3]
```diff
@@ -103,3 +101,6 @@ def __call__(
         if len(inferences) == 0:
             return None
         return inferences
+
+    def to(self, device: Device):
+        self._model.to(device=device.value)
```
Comment (CamiloInx) [inline vision_agent_tools/tools/owlv2.py:106]: @hrnn Nice! I think we have more models that we might need to add this method to. Could you please add them? (e.g florencev2, depth_anything_v2, ...)

File: vision_agent_tools/tools/shared_types.py (hunk 2 above)
Comment (hrnn) [reply vision_agent_tools/tools/shared_types.py:8]: Fixed

File: vision_agent_tools/tools/owlv2.py (hunk 3 above)
Comment (hrnn) [reply vision_agent_tools/tools/owlv2.py:117]: Fixed

**Review by camiloaz — APPROVED** [review]

**Pull request merged**